torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1555,9 +1555,25 @@ def sample_inputs_adjoint(self, device, dtype, requires_grad, **kwargs):
 def sample_inputs_T(self, device, dtype, requires_grad, **kwargs):
     make_arg = partial(make_tensor, dtype=dtype, device=device, requires_grad=requires_grad)
 
-    shapes = ((), (M, M))
+    shapes = ((), (M, M), (M, L))
     return (SampleInput(make_arg(shape)) for shape in shapes)
 
+def error_inputs_T(self, device):
+    make_arg = partial(make_tensor, device=device, dtype=torch.float32)
+
+    # Deprecated behavior in regular PyTorch, but throws an error in primTorch:
+    # https://github.com/pytorch/pytorch/issues/86968
+    if self.name == '_refs.T':
```

Comment:
I would add a kwarg to the error input function for whether this is an error or not, and then have the OpInfos `partial` the error input function with the kwarg set as expected, vs. adding control-flow based on the string name of the OpInfo